pytorch_lightning/loops/dataloader/evaluation_dataloader_loop.py

```diff
@@ -0,0 +1,268 @@
+from typing import Any, List, Optional, Sequence, Tuple, Union
+
+from deprecate.utils import void
+from torch.utils.data.dataloader import DataLoader
+
+import pytorch_lightning as pl
+from pytorch_lightning.loops.dataloader.dataloader_loop import DataLoaderLoop
+from pytorch_lightning.loops.evaluation_epoch_loop import EvaluationEpochLoop
+from pytorch_lightning.trainer.connectors.logger_connector.result import ResultCollection
+from pytorch_lightning.trainer.states import TrainerFn
+from pytorch_lightning.utilities.model_helpers import is_overridden
+from pytorch_lightning.utilities.types import EPOCH_OUTPUT, STEP_OUTPUT
+
+
+class EvaluationDataLoaderLoop(DataLoaderLoop):
+    """Loops over all dataloaders for evaluation."""
+
+    def __init__(self):
+        super().__init__()
+        self._max_batches: Optional[Union[int, Sequence[int]]] = None
+        self.outputs = []
+        self.evaluation_loop = EvaluationEpochLoop()
+
+        self._val_results = ResultCollection(training=False)
+        self._test_results = ResultCollection(training=False)
+
+    @property
+    def num_dataloaders(self) -> int:
+        """Returns the total number of dataloaders"""
+        return self._get_num_dataloaders(self.dataloaders)
+
+    @property
+    def dataloaders(self) -> Sequence[DataLoader]:
+        """Returns the validation or test dataloaders"""
+        if self.trainer.testing:
+            return self.trainer.test_dataloaders
+        return self.trainer.val_dataloaders
+
+    @property
+    def results(self) -> Optional[ResultCollection]:
+        """Returns the Results of obtained from all dataloaders"""
+        if self.trainer.validating or self.trainer.sanity_checking:
+            return self._val_results
+        elif self.trainer.testing:
+            return self._test_results
+        return None
+
+    @property
+    def predictions(self):
+        """Returns the predictions from all dataloaders"""
+        # TODO: fixme
```

Comment:
?

Reply:
yeah, now that we have a dataloader loop I wanted to make the inner loop that goes over the validation epoch collect predictions and then the dataloader loop combine them. 

Right now it's a bit weird that the inner loop collects all predictions across all dataloaders even though it only runs over one dataloader each run.